elide-core/src/main/java/com/yahoo/elide/core/pagination/Pagination.java

```diff
@@ -83,46 +91,57 @@ public static Pagination parseQueryParams(final MultivaluedMap<String, String> q
+                            // page[totals] is a valueless parameter, use value of 0 just so that its presence can
+                            // be recorded in the map
+                            pageData.put(paginationKey, 0);
                         }
-                    } else if (queryParamKey.equals(PAGE_TOTAL_KEY)) {
-                        final String value = paramEntry.getValue().get(0);
-                        if (value.equals("true")) {
-                            pageTotalsRequested.setTrue();
-                        } else if (value.equals("false")) {
-                            pageTotalsRequested.setFalse();
-                        } else {
-                            throw new InvalidValueException(PAGE_TOTAL_KEY + " must be either true or false");
+                        else {
+                            final String value = paramEntry.getValue().get(0);
+                            try {
+                                int intValue = Integer.parseInt(value, 10);
+                                if (paginationKey == PaginationKey.limit && intValue <= 0) {
+                                    throw new InvalidValueException(
+                                            "page[size] and page[limit] must contain positive values.");
+                                }
+                                pageData.put(paginationKey, intValue);
+                            } catch (ClassCastException e) {
+                                throw new InvalidValueException("page values must be integers");
+                            }
                         }
                     } else if (queryParamKey.startsWith("page[")) {
                         throw new InvalidValueException("Invalid Pagination Parameter. Accepted values are page[number]"
                                 + ",page[size],page[offset],page[limit],page[totals]");
                     }
                 });
+        return new Pagination(pageData).evaluate(null);
+    }
 
-        if (pageData.isEmpty()) {
-            return DEFAULT_PAGINATION;
-        }
+    /**
+     * Evaluates the pagination variables. Uses the Paginate annotation if it has been set for the entity to be
+     * queried.
+     * @param entityClass
+     */
+    public Pagination evaluate(final Class entityClass) {
+        Paginate paginate = entityClass != null ? (Paginate) entityClass.getAnnotation(Paginate.class) : null;
 
-        boolean isPageBased = pageData.containsKey(PaginationKey.page);
-        int limit = !pageData.containsKey(PaginationKey.limit) ? DEFAULT_PAGE_SIZE : pageData.get(PaginationKey.limit);
-        if (limit < 0) {
-            throw new InvalidValueException("page[size] and page[limit] must contain positive values.");
-        }
+        int defaultLimit = paginate != null ? paginate.defaultLimit() : this.DEFAULT_PAGE_SIZE;
+        int maxLimit = paginate != null ? paginate.maxLimit() : this.MAX_PAGE_SIZE;
```

Comment:
I'd drop `this` and use the naked variable name, or prefix with the class name if you prefer, since this is a class static.